app/helpers/application_helper.rb

```diff
@@ -288,34 +282,30 @@ def flot_bar_chart(name, xaxis_label, yaxis_label, data, options = {})
                 }.merge(options))
   end
 
-  def action_buttons(*args)
-    toolbar_action_buttons args
-  end
-
   def select_action_button(title, options = {}, *args)
     # the no-buttons code is needed for users with less permissions
-    return unless args
-    args = args.flatten.map{|arg| arg unless arg.blank?}.compact
-    return if args.length == 0
+    args = args.flatten.select(&:present?)
+    return if args.blank?
 
-    #single button
-    return content_tag(:span, args[0].html_safe, options.merge(:class=>'btn btn-default')) if args.length == 1
-
-    #multiple options
     content_tag(:div, options.merge(:class=>'btn-group')) do
-      link_to((title + " " + content_tag(:span, '', :class => 'caret')).html_safe, '#',
+      #single button
+      if args.length == 1
+        args[0].html_safe
+      #multiple options
+      else
+        link_to((title + " " + content_tag(:span, '', :class => 'caret')).html_safe, '#',
               :class => "btn btn-default dropdown-toggle", :'data-toggle' => 'dropdown') +
              content_tag(:ul, :class=>"dropdown-menu pull-right") do
                args.map{ |option| content_tag(:li,option) }.join(" ").html_safe
              end
+       end
     end
   end
 
-  def toolbar_action_buttons(*args)
```

Comment:
no point having two names for the same function, with one of them only used once...